viz/cmd/authz.go

```diff
@@ -89,62 +91,55 @@ func NewCmdAuthz() *cobra.Command {
 				os.Exit(1)
 			}
 			for _, server := range servers {
-				sazs, err := k8s.ServerAuthorizationsForServer(cmd.Context(), k8sAPI, options.namespace, server)
+				requestParams := util.StatsSummaryRequestParams{
+					StatsBaseRequestParams: util.StatsBaseRequestParams{
+						TimeWindow:    options.timeWindow,
+						ResourceName:  server,
+						ResourceType:  k8s.Server,
+						Namespace:     options.namespace,
+						AllNamespaces: false,
+					},
+				}
+
+				req, err := util.BuildStatSummaryRequest(requestParams)
+				if err != nil {
+					return err
+				}
+				resp, err := requestStatsFromAPI(client, req)
 				if err != nil {
 					fmt.Fprint(os.Stderr, err.Error())
 					os.Exit(1)
 				}
-				for _, saz := range sazs {
-					requestParams := util.StatsSummaryRequestParams{
-						StatsBaseRequestParams: util.StatsBaseRequestParams{
-							TimeWindow:    options.timeWindow,
-							ResourceName:  saz,
-							ResourceType:  k8s.ServerAuthorization,
-							Namespace:     options.namespace,
-							AllNamespaces: false,
-						},
-						ToNamespace: options.namespace,
-					}
-					requestParams.ToName = server
-					requestParams.ToType = k8s.Server
 
-					req, err := util.BuildStatSummaryRequest(requestParams)
-					if err != nil {
-						return err
+				for _, row := range respToRows(resp) {
+					var authzp, saz, route string
+					if row.GetSrvStats().GetAuthz() != nil {
+						if row.GetSrvStats().GetAuthz().Type == "authorizationpolicy" {
+							authzp = row.GetSrvStats().GetAuthz().Name
+						} else if row.GetSrvStats().GetAuthz().Type == "serverauthorization" {
+							saz = row.GetSrvStats().GetAuthz().Name
+						}
```

Comment:
nit, take it or leave it: is there a reason we don't write something like:
```suggestion
                    authz := row.GetSrvStats().GetAuthz()
					if authz != nil {
						if authz.Type == "authorizationpolicy" {
							authzp = authz.Name
						} else if authz.Type == "serverauthorization" {
							saz = authz.Name
						}
```

Reply:
i guess all this saves us is nil-checking the `ServerStats` every time?